relay-config/src/config.rs

```diff
@@ -643,6 +700,160 @@ impl Config {
+                processing.kafka_config.push(KafkaConfigParam {
+                    name: "bootstrap.servers".to_owned(),
+                    value: kafka_url,
+                })
+            }
+        }
+        // credentials overrides
+        let id = if let Some(id) = overrides.id {
+            let id = Uuid::parse_str(&id).map_err(|err| {
+                ConfigError::new_field_error(
+                    "id",
+                    id,
+                    err.context(ConfigErrorKind::InvalidOverride),
+                )
+            })?;
+            Some(id)
+        } else {
+            None
+        };
+        let public_key = if let Some(public_key) = overrides.public_key {
+            let public_key = PublicKey::from_str(&public_key).map_err(|err| {
+                ConfigError::new_field_error(
+                    "public_key",
+                    public_key,
+                    err.context(ConfigErrorKind::InvalidOverride),
+                )
+            })?;
+            Some(public_key)
+        } else {
+            None
+        };
+
+        let secret_key = if let Some(secret_key) = overrides.secret_key {
+            let secret_key = SecretKey::from_str(&secret_key).map_err(|err| {
+                ConfigError::new_field_error(
+                    "secret_key",
+                    secret_key,
+                    err.context(ConfigErrorKind::InvalidOverride),
+                )
+            })?;
+            Some(secret_key)
+        } else {
+            None
+        };
+
+        if let Some(credentials) = &mut self.credentials {
+            //we have existing credentials we may override some entries
+            if let Some(id) = id {
+                credentials.id = id;
+            }
+            if let Some(public_key) = public_key {
+                credentials.public_key = public_key;
+            }
+            if let Some(secret_key) = secret_key {
+                credentials.secret_key = secret_key
+            }
+        } else {
+            //no existing credentials we may only create the full credentials
+            match (id, public_key, secret_key) {
+                (Some(id), Some(public_key), Some(secret_key)) => {
+                    self.credentials = Some(Credentials {
+                        id,
+                        public_key,
+                        secret_key,
+                    })
+                }
+                (None, None, None) => {
+                    // nothing provided, we'll just leave the credentials None, maybe we
+                    // don't need them in the current command or we'll override them latter
+                }
+                (id, public_key, secret_key) => {
+                    return Err(ConfigError::new_field_error(
+                        "incomplete credentials",
+                        format!(
+                            "id={:?} public_key={:?} secret_key={:?}",
+                            id, public_key, secret_key
```

Comment:
I think this is pretty excessive. It would be nicer to just print out which field (e.g. relay id, public key, ...) is missing, but not the value. Actually, in all cases.

Reply:
ok